tests/integration/test_fim/test_registry/test_restrict/test_registry_restrict.py

```diff
@@ -0,0 +1,194 @@
+        assert event['data']['arch'] == '[x32]' if arch == KEY_WOW64_32KEY else '[x64]', f'Key event arch not equal'
+
+        while True:
+            ignored_value = wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                                                   callback=callback_restricted,
+                                                   error_message='Did not receive expected '
+                                                                 '"Sending FIM event: ..." event').result()
+            if ignored_value == value_name:
+                break
+
+    delete_registry_value(key_h, value_name)
+    # Go ahead in time to let syscheck perform a new scan
+    check_time_travel(get_configuration['metadata']['fim_mode'] == 'scheduled', monitor=wazuh_log_monitor)
+    event = wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                                callback=callback_detect_event, accum_results= 2 if triggers_event else 1).result()
+
+    if triggers_event:
+        assert event[0]['data']['type'] == 'modified', f'Key event not modified'
+        assert event[0]['data']['path'] == os.path.join(key, subkey), f'Key event wrong path'
+        assert event[0]['data']['arch'] == '[x32]' if arch == KEY_WOW64_32KEY else '[x64]', f'Key event arch not equal'
+
+        assert event[1]['data']['type'] == 'deleted', f'Event type not equal'
+        assert event[1]['data']['path'] == os.path.join(key, subkey), f'Event path not equal'
+        assert event[1]['data']['value_name'] == value_name, f'Value name not equal'
+        assert event[1]['data']['arch'] == '[x32]' if arch == KEY_WOW64_32KEY else '[x64]', f'Value event arch not equal'
+    else:
+        # After deleting the value, we don't expect any message of the value because it's not in the DB
+        assert event['data']['type'] == 'modified', f'Key event not modified'
+        assert event['data']['path'] == os.path.join(key, subkey), f'Key event wrong path'
+        assert event['data']['arch'] == '[x32]' if arch == KEY_WOW64_32KEY else '[x64]', f'Key event arch not equal'
+
+
+@pytest.mark.parametrize('key, subkey, test_subkey, arch, triggers_event, tags_to_apply', [
+    (key, sub_key_1, valid_subkey, KEY_WOW64_64KEY, True, {'key_restrict'}),
+    (key, sub_key_2, valid_subkey, KEY_WOW64_32KEY, True, {'key_restrict'}),
+    (key, sub_key_2, valid_subkey, KEY_WOW64_64KEY, True, {'key_restrict'}),
+    (key, sub_key_1, no_valid_subkey, KEY_WOW64_64KEY, False, {'key_restrict'}),
+    (key, sub_key_2, no_valid_subkey, KEY_WOW64_32KEY, False, {'key_restrict'}),
+    (key, sub_key_2, no_valid_subkey, KEY_WOW64_64KEY, False, {'key_restrict'})
+    ])
+def test_restrict_key(key, subkey, test_subkey, arch, triggers_event, tags_to_apply,
+                 get_configuration, configure_environment, restart_syscheckd,
+                  wait_for_fim_start):
+    """
+    Check the only files detected are those matching the restrict regex
```

Comment:
This test is about registry keys, not files. Please, change it.
```suggestion
    Check the only registry keys detected are those matching the restrict regex.
```